src/main/java/com/algomart/kibouregistry/services/implementations/ParticipantsServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.algomart.kibouregistry.services.implementations;
+
+import com.algomart.kibouregistry.dao.ParticipantsRepo;
+import com.algomart.kibouregistry.entity.Participants;
+import com.algomart.kibouregistry.exceptions.ResourceNotFoundException;
+import com.algomart.kibouregistry.services.ParticipantsService;
+import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class ParticipantsServiceImpl implements ParticipantsService {
+
+    private final ParticipantsRepo participantsRepo;
+
+    @Override
+    @Transactional
+    @PreAuthorize("hasRole('ADMIN')")
+    public Participants addParticipant(Participants participant) {
+        return participantsRepo.save(participant);
+    }
+
+    @Override
+    @PreAuthorize("hasRole('ADMIN')")
+    public List<Participants> getAllParticipants() {
+        List<Participants> participantsList = participantsRepo.findAll();
+        if (participantsList.isEmpty()) {
+            throw new ResourceNotFoundException("No participant was found.");
+        }
+        return participantsRepo.findAll();
+    }
+
+    @Override
```

Comment:
For all your methods, return proper response

Reply:
I thought that should be done under controller

Reply:
I don't get that of service. Proper responses, how?

Reply:
You should have better-structured responses. You and I already discussed this before on a call. Work on that